Makefile

```diff
@@ -30,35 +30,16 @@ DEVWORKSPACE_CTRL_SA=devworkspace-controller-serviceaccount
 INTERNAL_TMP_DIR=/tmp/devworkspace-controller
 BUMPED_KUBECONFIG=$(INTERNAL_TMP_DIR)/kubeconfig
 RELATED_IMAGES_FILE=$(INTERNAL_TMP_DIR)/environment
+GO_PACKAGE_PATH ?= $(shell head -n1 go.mod | cut -d " " -f 2)
+BUILD_TIME = $(shell date -u '+%Y-%m-%dT%H:%M:%SZ')
+REF = $(shell cat .git/HEAD | sed 's|ref: ||')
+GIT_COMMIT_ID := $(shell cat .git/${REF})
 
-ifeq (,$(shell which kubectl))
-ifeq (,$(shell which oc))
-$(error oc or kubectl is required to proceed)
-else
-K8S_CLI := oc
-endif
-else
-K8S_CLI := kubectl
-endif
-
-ifeq ($(shell $(K8S_CLI) api-resources --api-group='route.openshift.io'  2>&1 | grep -o routes),routes)
-PLATFORM := openshift
-else
-PLATFORM := kubernetes
-endif
-
-# minikube handling
-ifeq ($(shell $(K8S_CLI) config current-context 2>&1),minikube)
-# check ingress addon is enabled
-ifeq ($(shell minikube addons list -o json | jq -r .ingress.Status), disabled)
-$(error ingress addon should be enabled on top of minikube)
-endif
-export ROUTING_SUFFIX := $(shell minikube ip).nip.io
+ifneq (,$(shell which kubectl 2>/dev/null)$(shell which oc 2>/dev/null))
+include deploy.mk
```

Comment:
Splitting the file in two breaks our `make help` functionality :(

Line 269 in `Makefile` should be updated to
```bash
	@sed -n 's/^### /    /p' $< deploy.mk | awk 'BEGIN { FS=":" } { printf "%-30s -%s\n", $$1, $$2 }'
```

Reply:
@amisevsk yeahh.. i was thinking to update it to be like this: https://github.com/kwk/llvm-ci/blob/ffa5ce5245/infra/help.mk#L10-L35 - I've tested it and it seems to work! WDYT?

Reply:
Actually for now, maybe lets just go with adding deploy.mk to the help target. We can think about updating the help if we need to later on! I've updated to add deploy.mk!